helpers/scrolling.ts

```diff
@@ -139,18 +145,52 @@ export function scrollHorizontally(
   if (left - parentLeft < opts.leftOffset) {
     const goal = scroll.x - opts.leftOffset + left - parentLeft;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 
   if (right - parentLeft > width - opts.rightOffset) {
     const goal = scroll.x + right - parentLeft - width + opts.rightOffset;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 }
+
+type ScrollFunction = (element: Element | Window, x: number, y?: number) => Promise<void>;
+
+type ScrollQueueItem = {
```

Comment:
How do you choose between a type and an interface ? Why type instead of interface here for exemple ?

Reply:
Ha! Good question :-). I don't know. I guess I started with something much simpler, and just added to it, keeping `type`. I don't have any rule for choosing one or the other. I guess `interface` is just as appropriate here.

Anybody else have thoughts about this?

Reply:
OK, will make `ScrollQueueItem` an interface 👍 .